drltrace/scripts/gen_drlttrace_config.py

```diff
@@ -0,0 +1,379 @@
+"DWORDLONG":"DWORDLONG","EXCEPTION_POINTERS":"EXCEPTION_POINTERS", "_EXCEPTION_POINTERS":"_EXCEPTION_POINTERS",
+"WOW64_CONTEXT":"WOW64_CONTEXT", "UILANGUAGE_ENUMPROCW":"UILANGUAGE_ENUMPROCW",
+"UILANGUAGE_ENUMPROCA":"UILANGUAGE_ENUMPROCA", "TIMEFMT_ENUMPROCW":"TIMEFMT_ENUMPROCW",
+"TIMEFMT_ENUMPROCEX":"TIMEFMT_ENUMPROCEX", "TIMEFMT_ENUMPROCA":"TIMEFMT_ENUMPROCA",
+"NUMBERFMTW":"NUMBERFMTW",  "LOGICAL_PROCESSOR_RELATIONSHIP":"LOGICAL_PROCESSOR_RELATIONSHIP",
+"LOCALE_ENUMPROCW":"LOCALE_ENUMPROCW", "LOCALE_ENUMPROCEX":"LOCALE_ENUMPROCEX",
+"LOCALE_ENUMPROCA":"LOCALE_ENUMPROCA", "LATENCY_TIME":"LATENCY_TIME", "LANGUAGEGROUP_ENUMPROCW":"LANGUAGEGROUP_ENUMPROCW",
+"LANGUAGEGROUP_ENUMPROCA":"LANGUAGEGROUP_ENUMPROCA", "GROUP_AFFINITY":"GROUP_AFFINITY", "GEO_ENUMPROC":"GEO_ENUMPROC",
+"EXECUTION_STATE":"EXECUTION_STATE", "DYNAMIC_TIME_ZONE_INFORMATION":"DYNAMIC_TIME_ZONE_INFORMATION",
+"CURRENCYFMTW":"CURRENCYFMTW", "CONTEXT":"CONTEXT", "CONSOLE_CURSOR_INFO":"CONSOLE_CURSOR_INFO", "CHAR":"char",
+"APPLICATION_RECOVERY_CALLBACK":"APPLICATION_RECOVERY_CALLBACK", "HRESULT": "HRESULT", "PCNZCH": "char*",
+"LPPROCESS_INFORMATION": "PROCESS_INFORMATION*", "LPINT": "int*", "LSTATUS":"LSTATUS",
+"DEP_SYSTEM_POLICY_TYPE":"DEP_SYSTEM_POLICY_TYPE", "CURRENCYFMTA":"CURRENCYFMTA", "va_list": "va_list"}
+
+
+tokens_ignore_prefix = {"WINAPI", "WINBASEAPI", "APIENTRY", "virtual", "DECLSPEC_NORETURN",
+                        "WINNORMALIZEAPI"}
+
+PATH = "results/" # path where the output of headers_parser.py was saved
+
+unknown_types = dict() # we use this dictionary to save and print a list of unknown types
+
+def parse_internal_parentesses(str):
+    while True:
+        end = str.find(")")
+        if end == -1:
+            break
+        start = str.rfind("(", 0, end)
+        if start == -1:
+            break
+        new_str = str[:start]
+        new_str += str[end+1:]
+        str = new_str
+    return str
+
+def get_type(type_str):
+    return str(types_map.get(type_str, None))
+
+def parse_prefix(line):
+    '''
+    The function searches for a WinAPI return type in the string specified
+    @in line - a string with a type to find
+    @out - returning type
+    '''
+    if "//" in line or "Boolean" in line or "&&" in line:
+        return None
+    if ")" in line:
+        line = parse_internal_parentesses(line)
+
+    line = line.split(" ")
+    func_type = None
+    has_type = 0
+    ignore_until_paren = False
+    for element in line:
+        if "__success" in element:
+            ignore_until_paren = True
+        if ")" in element:
+            ignore_until_paren = False
+            continue
+        if ignore_until_paren == True:
+            continue
+        if element in tokens_ignore_prefix or element == "" or "__" in element\
+           or (("WIN" in element or "NT" in element) and "API" in element):
+            continue
+        if has_type >= 1:
```

Comment:
"has_type" sounds like a boolean: instead it looks like it's meant to be a count of tokens before the first '('?  (It won't work for multiple () as mentioned above.)  How about "type_token_count" instead as a name?

But won't it include the function name too?  It seems like it would always come here and fail.